test/basicfuns.jl

```diff
@@ -44,6 +44,34 @@
     @test iszero(xlog1py(0 + im * 0, -1 + im * Inf))
 end
 
+@testset "xexpx" begin
+    for z in (0, 0.0, 0f0, false, -Inf)
+        @test iszero(@inferred xexpx(z))
+    end
+    for nan in (NaN16, NaN32, NaN64)
+        @test isnan(@inferred xexpx(nan))
+    end
+    for inf in (Inf16, Inf32, Inf64)
+        @test inf === @inferred xexpx(inf)
+    end
+    for u in (1, true, 1.0, 1f0)
+        @test exp(u) === @inferred xexpx(u)
+    end
+    @test 2exp(2) ≈ @inferred xexpx(2)
+end
+
+@testset "xexpy" begin
+    @test iszero(@inferred xexpy(Inf, -Inf))
+    @test isnan(@inferred xexpy(NaN, -Inf))
+    @test isnan(@inferred xexpy(NaN, 1))
+    @test isnan(@inferred xexpy(0, NaN))
+    @test iszero(@inferred xexpy(0, -Inf))
+    @test iszero(@inferred xexpy(1., -Inf))
```

Comment:
I guess it might be clearer to reorder the tests. Additionally, I suggest adding tests with other types, e.g.,
```suggestion
    for x in (-Inf, -Inf32)
        @test @inferred(xexpy(0, x)) === zero(x)
        @test @inferred(xexpy(one(x), x)) === zero(x)
        @test @inferred(xexpy(x, x)) === zero(x)
        @test @inferred(xexpy(-x, x)) === zero(x)
    end

    for x in (NaN, NaN32)
        @test @inferred(xexpy(x, oftype(x, -Inf))) === x
        @test @inferred(xexpy(x, 1)) === x
        @test @inferred(xexpy(x, Inf32)) === x
        @test @inferred(xexpy(-Inf32, x)) === x
        @test @inferred(xexpy(0, x)) === x
        @test @inferred(xexpy(oftype(x, Inf), x)) === x
        @test @inferred(xexpy(x, x)) === x
    end
```